src/main/java/duke/task/Deadline.java

```diff
@@ -0,0 +1,75 @@
+package duke.task;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+/**
+ * Represents the deadline type task that has a deadline date and an optional parsedDate
+ * if provided date string is parsable
+ */
+public class Deadline extends ListItem {
+    private final String date;
+    private LocalDate parsedDate;
+
+    /**
+     * Constructor for Deadline that was not provided the task done status
+     * @param task takes in a string and pass to parent's constructor as task name
+     * @param inputDate the deadline input by user
+     */
+    public Deadline(String task, String inputDate) {
+        super(task);
+        this.date = inputDate;
+        this.parsedDate = parseDate(inputDate);
+    }
+
+    /**
+     * the overloaded constructor that allows taking the status of the task
```

Comment:
Not sure if you meant to not capitalize the "the over..." 
and also perhaps across classes, might want to standardize the wording for overloaded constructors (eg. with Event class)